### DISCUSSION TITLE
Errore installazione python 3.9.7

### DISCUSSION THREAD

**Discussion by Topolynx** [indomus/forum#1137] [HUB: Home Assistant]
Ciao a tutti, mi potreste aiutare?
Sto cercando di aggiornare python alla versione 3.9.7 ma ho dei messaggi di avvertimento
```
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/10151>
distutils: /usr/local/include/python3.9/UNKNOWN
sysconfig: /home/pi/Python-3.9.7/Include/UNKNOWN
WARNING: Additional context:
user = False
home = None
root = '/'
prefix = None
Looking in links: /tmp/tmpfhi8sun_
Requirement already satisfied: setuptools in /usr/local/lib/python3.9/site-packages (57.4.0)
Requirement already satisfied: pip in /usr/local/lib/python3.9/site-packages (21.2.3)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```
Rispetto alla guida su indomus, siccome al tempo ebbi dei problemi di installazione,  usai due diverse opzioni per l'installazione della versione 3.8.6 ovvero
```
sudo make -j 4
sudo make altinstall
```
e poi avevo impostato l'ultima versione come default:

```
echo "alias python=/usr/local/bin/python3.8" >> ~/.bashrc
source ~/.bashrc
```
Mi potete aiutare a capire cosa sta accadendo per favore?
Grazie a tutti!

**Comment by snakuzzo** [comment]
Premesso che...

- di fatto non aggiorni Python, ma ne installi una nuova versione che si affianca a quella che hai già
- le operazioni che dici di aver fatto "in più" non hanno nulla a che vedere con il problema che stai riscontrando
- quel problema non mi pare relativo all'installazione della nuova versione di Python (che dovresti già aver installato), ma a `pip`

...leggi il primo messaggio di warning...e contorlla quella issue che viene indicata. E' ancora aperta su github ed è relativa a `pip 21.2`

**Comment by Topolynx** [comment]
Pardon, @snakuzzo, c'è una release di Python 3.9 in particolare che mi consigli di installare per Home Assistant?

**Reply by snakuzzo** [reply]
Una vale l'altra

**Reply by Topolynx** [reply]
grazie!
